schema/deploy/shipping/views.sql

```diff
@@ -2161,16 +2204,18 @@ create or replace view shipping.return_results_v3 as
             when sample_id is null then 'not-received'
             when never_tested then 'never-tested'
             when sample_id is not null and presence_absence_id is null then 'pending'
-            when hcov19_present is true then 'positive'
-            when hcov19_present is false then 'negative'
-            when presence_absence_id is not null and hcov19_present is null then 'inconclusive'
+            when hcov19_pa.present is true then 'positive'
+            when hcov19_pa.present is false then 'negative'
+            when presence_absence_id is not null and hcov19_pa.present is null then 'inconclusive'
         end as status_code,
         result_ts,
         swab_type,
         staff_observed
     from
       samples
-      left join hcov19_presence_absence using (sample_id)
+      left join shipping.hcov19_presence_absence_result_v1 as hcov19_pa using (sample_id)
+    where
+      hcov19_pa.details is null or hcov19_pa.details @> '{"assay_type": "Clia"}'
```

Comment:
I think this subtly changes the rows retrieved by the join.

In the existing version, rows with a null `details` would be omitted because the `assay_type` check (in the `hcov19_presence_absence` CTE) would evaluate to null.

In the new version, they're included because of the `is null` clause here.

I think the intent was to not require the `assay_type` check if the left join produced no rows. In that case, the correct condition is to check if the join key is null, e.g.:

```sql
hcov19_pa.sample_id is null
```

Reply:
Thanks for catching that, Tom! I just re-pushed with `hcov19_pa.sample_id is null` replacing `hcov19_pa.details` is null. Please take a look when you get a chance to make sure I understood your feedback. 

A quick count of total rows returned in each view shows no regressions. I am going to continue doing a bit more testing, though, to be safe.

Reply:
Yep, that's what I meant. Thanks!

For testing things like this, which are intended to be non-changes, I like to dump the whole darn view to a file before and after, and then diff them to confirm no changes with existing data.

Reply:
Ah, that's the most robust testing strategy of all! I thought to do that at one point but apparently forgot.

Reply:
I just did a `diff` of the three shipping views that use the refactored hcov19 p/a view, and there were no changes :+1: